### PR TITLE
Enable prevention of info file generation (aides #2)

### PR DESCRIPTION
**Pull request by Eagle3386**
I neither take copyright nor liability for the code contained within this PR, except mentioning of my full name & link to my GH profile as contained within this PR.

### CODE REVIEW

**Review by Eagle3386 — COMMENTED** [review]
Both directives create IntelliSense warnings & are _always_ ignored since this T4 script will be included (which its file extension, `.ttinclude` already suggests) via `include` directive in a `.tt` or even another `.ttinclude` file.
Actual IntelliSense warnings:
- output: `Multiple template directives were found in the template. All but the first one will be ignored. Multiple parameters to the template directive should be specified within one template directive`
- template: `Multiple template directives were found in the template. All but the first one will be ignored. Multiple parameters to the template directive should be specified within one template directive.`

Additionally changes:
- fixed XML docs
- sorted "usings" according to Microsoft code guidelines (`System.*`, then 3rd party - all alphabetically) & removed 1 obsolete
- refactored several methods to be simplified & use modern C# syntactic sugar where possible
- removed trailing whitespaces

**Comment by jgraber** [comment]
Hi Martin,
Thanks for the pull request. Unfortunately, there are multiple problems with your changes that makes it impossible for us to accept this pull request.

First, and most important for us, your change breaks 8 of our 11 features of the T4.FileManager:
image

Second, your pull request contains multiple changes around XML documentation that we do not want. Next time please ask if it is ok to go and modify formatting rules in a project before you change them.

Third, there are some logical bugs (like = in a comparison instead of ==) that should be detected by tests - and they are missing in your pull request.

By reading your comment to the pull request we did not understand what problem you try to solve in the first place. In our understanding your change breaks the basic concept of T4.FileManager. If you are still interested to contribute, please contact us at connect@databinding.gmbh to first check if this change fits in our vision.

We will add our feature files in the next weeks to get the build pipeline up on GitHub so that everyone gets faster feedback and can check their changes against our feature list. 

Regards,
Johnny & René

**Comment by Eagle3386** [comment]
> Hi Martin,

Hi, Johnny & René!

> Thanks for the pull request. Unfortunately, there are multiple problems with your changes that makes it impossible for us to accept this pull request.

I'd happily help you resolve them, so let's get straight into it:

> First, and most important for us, your change breaks 8 of our 11 features of the T4.FileManager:
> image

tl;dr: Adopting my changes into the test file's fragments will resolve this quickly.

As you've already stated yourself, they all broke due to the tests not being publicly available which in turn made it impossible for me to even know about or how to adopt them accordingly.
Since then, you've made them available & from what I see, those breaking do so mostly because of featuring (part of) the T4 file's content, i. e. there's no actual code validation (e. g. code does what's supposed to do), but rather a code matching validation (e. g. code fragment in test file matches its pendant fragment within the "real" source).

> Second, your pull request contains multiple changes around XML documentation that we do not want. Next time please ask if it is ok to go and modify formatting rules in a project before you change them.

For this, I _purposely_ did them all in a single, stand-alone commit, ultimately named "Refactor file layout towards MS code guidelines" - which gave you all the good stuff at once:
1. Accept the actual improvements, but reject the refactoring.
2. See for yourself what's currently wrong: `import`s are non-alphabetical (same goes for members & properties within the classes), non-"System.* first, then 3rd party, then your own" & the original `template` directive in line 1 actually produces a warning in consuming T4 scripts which in turn makes my refactoring towards that directive more a bug fix than a mere refactoring.
3. No developer likes scrolling useless lines - which each & every `///<summary>` line that is actually short enough to be one-liner breaks. But as I've said in this paragraph's beginning, it's _intentionally_ in 1 commit so that you guys could choose to omit it while picking one (or all) other commits from my PR.
4. Not having talked/written to you prior to my PR is something I could've done better back then, but I looked at the repo's recent activity (& @db-ty's, too) & since there was none, I believed you prefer "less communication, more PR". I apologize for this mistake.

> Third, there are some logical bugs (like = in a comparison instead of ==) that should be detected by tests - and they are missing in your pull request.

Regarding the "missing tests", I already explained that above.
Regarding the "logical bugs", I'd like to ask for at least one example since - even after looking at my changes again - I couldn't find such a mistake.
If you're suspecting line 47 (`if (!(IsInfoFileGenerationEnabled = isInfoFileGenerationEnabled))`) to be such a "bug", it's just syntactic sugar for "assign method parameter value to property, only then evaluate" as I'm a big fan of such concise statements instead of its boilerplate-like alternative:
```csharp
IsInfoFileGenerationEnabled = isInfoFileGenerationEnabled;
if (!IsInfoFileGenerationEnabled)
```
> By reading your comment to the pull request we did not understand what problem you try to solve in the first place. In our understanding your change breaks the basic concept of T4.FileManager. If you are still interested to contribute, please contact us at [connect@databinding.gmbh](mailto:connect@databinding.gmbh) to first check if this change fits in our vision.

That's completely explained in the referenced issue (#2), even hinted to in the PR's title. I'm wondering, if you actually read my comments or only thought "tests failed, file reformatted - let's skip this BS".. 😞

Put simply, my PR overcomes 2 things:
1. When used for generation of independent C# code files, Visual Studio also **always** creates an empty `<T4ScriptFileName>.cs` file, modifying its corresponding `.csproj` file due to the empty file being added, again & again - which is simply annoying to any developer using any version control system.
2. In case the T4 script fails, its `<T4ScriptFileName>.cs` output file only contains `ErrorGeneratingOutput`, essentially breaking the build of its corresponding project due to `.cs` files being adding with their `Build action` set to `Compile` - which is even worse whenever the T4 script is contained within the same project as it creates kinda chicken-&-egg problem, requiring to deletion of the `<T4ScriptFileName>.cs` file, rebuilding the project & re-running the T4 script to finally achieve the desired goal.

> We will add our feature files in the next weeks to get the build pipeline up on GitHub so that everyone gets faster feedback and can check their changes against our feature list.

Thanks for that. However, I do like to re-request my PR getting merged due to the fact that I purposely allowed you, the maintainers, to modify my PR to your liking - you could leave out that "XML docs commit" & still merge my improvement.
Besides, I don't really like your BDD-based approach via SpecFlow. To me, it's neither TDD nor at least unit tests. However, the latter is what's widely common sense at GitHub for any PR.

Maybe you want to reconsider that & slice that BDD part out of the code & test stuff of the repo - in that case, I'd happily add unit tests to my PR. 🙂

> Regards,
> Johnny & René

Kind regards & stay safe,
 Martin
